exchanges/account/account.go

```diff
@@ -268,22 +275,31 @@ func (s *Service) Update(incoming *Holdings, creds *Credentials) error {
 
 // load checks to see if there is a change from incoming balance, if there is a
 // change it will change then alert external routines.
-func (b *ProtectedBalance) load(change Balance) {
+func (b *ProtectedBalance) load(change *Balance) error {
+	if change == nil {
+		return fmt.Errorf("%w for '%T'", common.ErrNilPointer, change)
+	}
 	b.m.Lock()
 	defer b.m.Unlock()
+	if !b.updatedAt.IsZero() && !b.updatedAt.Before(change.UpdatedAt) {
+		return errOutOfSequence
+	}
 	if b.total == change.Total &&
```

Comment:
While I appreciate the added field, looking across all usages of `Balance`, no exchange populates the field `UpdatedAt`.
You could do something as simple as
`if change.UpdatedAt.IsZero() { change.UpdatedAt = time.Now() }`
Or you could populate it at all the sources

Reply:
Fixed it by https://github.com/thrasher-corp/gocryptotrader/pull/1827/commits/fa0651d8341cb22f53ff8154d1e991ce6d5b8732